applications/asset_tracker_v2/tests/cloud_codec_nrf_mocked_cjson/src/nrf_cloud_codec_mocked_test.c

```diff
@@ -0,0 +1,576 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdbool.h>
+
+#include "cloud_codec.h"
+#include <cJSON_os.h>
+#include <errno.h>
+
+#include "mock_json_helpers.h"
+#include "mock_date_time.h"
+#include "mock_cJSON_to_mock.h"
+#include "float.h"
+
+
+struct cloud_codec_data codec;
+int ret;
+
+void setUp(void)
+{
+	__wrap_cJSON_Init_Expect();
+	ret = cloud_codec_init(NULL, NULL);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	memset(&codec, 0, sizeof(codec));
+}
+
+void tearDown(void)
+{
+}
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+/**
+ * @brief mocking-fueled tests for nrf cloud codec
+ *
+ * The following tests focus on covering error handling.
+ * There is special consideration for heap objects to detect irregularities
+ * wrt. allocation/deallocation.
+ * The comments on the right side track lifetimes of cJSON objects.
+ *
+ */
+
+void test_enc_ui_fail0(void)
```

Comment:
A one liner explanation of the tests will be useful

Reply:
For these tests, these explanations would be very hard to write. All of them force some internal function to fail at specific points in execution. Those points are precisely laid out in the comments inside the functions.

Reply:
Then perhaps you can mention that somewhere ? Or perhaps you have already done so and I missed to see it :D

Reply:
I added a little comment about that around line 50 now.